### PR TITLE
[KEYCLOAK-16611] Add listType to realm/RedirectorIdentityProviderOverride

### PR DESCRIPTION
**Pull request by dhx-teuto**
https://issues.redhat.com/browse/KEYCLOAK-16611

This prevents a warning while running make code/gen.

## Additional Information
https://github.com/kubernetes/kube-openapi/blob/master/pkg/idl/listtype_test.go

### CODE REVIEW

File: pkg/apis/keycloak/v1alpha1/keycloakrealm_types.go [hunk 1]
```diff
@@ -18,6 +18,8 @@ type KeycloakRealmSpec struct {
 	// +kubebuilder:validation:Required
 	Realm *KeycloakAPIRealm `json:"realm"`
 	// A list of overrides to the default Realm behavior.
+	// +listType=map
+	// +listMapKey=identityProvider
```
Comment (slaskawi) [inline pkg/apis/keycloak/v1alpha1/keycloakrealm_types.go:22]: A better solution is to make this `+listType=atomic`. We don't really care about merging this list.

**Review by slaskawi — CHANGES_REQUESTED** [review]

**Comment by slaskawi** [comment]
@dhx-teuto Could you please create a JIRA for this?

**Comment by slaskawi** [comment]
I got notified from other channels that this might be an important issue. Could you please address all the comments I mentioned, @dhx-teuto ? I would love to get this in quickly.

File: pkg/apis/keycloak/v1alpha1/keycloakrealm_types.go (hunk 1 above)
Comment (dhx-teuto) [reply pkg/apis/keycloak/v1alpha1/keycloakrealm_types.go]: Done.

**Comment by dhx-teuto** [comment]
> I got notified from other channels that this might be an important issue. Could you please address all the comments I mentioned, @dhx-teuto ? I would love to get this in quickly.

@slaskawi Sorry I missed your comment. I created the Jira issue and changed it to atomic.

File: pkg/apis/keycloak/v1alpha1/keycloakrealm_types.go (hunk 1 above)
Comment (slaskawi) [reply pkg/apis/keycloak/v1alpha1/keycloakrealm_types.go]: Thank you!

**Review by slaskawi — APPROVED** [review]

**Comment by slaskawi** [comment]
Integrated, thanks @dhx-teuto

**Pull request merged**